snap7/client.py

```diff
@@ -500,7 +504,12 @@ def as_db_get(self, db_number):
         check_error(result, context="client")
         return bytearray(_buffer)
 
-    def as_db_read(self, db_number, start, size):
+    async def async_wait_loop(self):
```

Comment:
I think we should move the python async functions to a separate module and keep access to the lower level functions also. So just leave the function as is, but put the python asyncio specific code into a separate submodule. What do you think?

Reply:
I can do that if you wish, but I don't know if the "old" wrapper would work properly, since nobody is checking if a result already returned and the actual code doesn't allow to check in one of the three ways that exist.

Even then, it would be still the same if I poll all the time or wait idle_time until a value is received. Only the Callback way would work, which would end in future objects (and more confusing I guess).

Reply:
I would add an optional parameter, which could be used to choose different ways of checking if a value received.
Standard would be `None ` like "use original code".

Reply:
I think it is just much more clear if there is one module that is a one-to-one direct wrapper to the snap7 lib and then a separate submodule that does the fancy new asyncio stuff. Maybe people actually use those calls already and build their own logic around it, so there is also a backwards compatibility argument there.

Reply:
I have chosen to create another class in client.py for two reasons:
1) It is still a Client, so intuitive it could join client.py as well
2) The async Client should be a childclass from Client, to be (and stay) on the same level. The necessarry methods are overwritten.
3) I was afraid if I import `Client` and some dependencies may change in future, that these are not imported in the seperate module. I don't want to `import client *`. A linter may not even recognise that dependencies are missing.